src/dirdef.cpp

```diff
@@ -1096,3 +1093,17 @@ void generateDirDocs(OutputList &ol)
   }
 }
 
+bool compareDirDefs(const DirDef *item1, const DirDef *item2)
+{
+  return qstricmp(item1->shortName(),item2->shortName()) < 0;
+}
+
+void sortInDirList(DirList& list, DirDef *const newItem)
+{
+  auto potentialSuccessor = list.begin();
+  while (potentialSuccessor != list.cend() && compareDirDefs(*potentialSuccessor, newItem))
+  {
+    potentialSuccessor++;
+  }
+  list.insert(potentialSuccessor, newItem);
+}
```

Comment:
I performed a [test of the correct integration of `compareDirDefs` and `sortInDirList`](https://github.com/doxygen/doxygen/pull/7728#issuecomment-622994606) by turning [this `<` operator](https://github.com/doxygen/doxygen/pull/7728/files#diff-c0b61dd85e8f538c4e4e369f0e52cb4cR1098) and checked that the output in the generated documentation was in reverse order.

- [used test configuration](https://github.com/doxygen/doxygen/files/4605861/test_sortInDirList_compareDirDefs.zip)